CMakeLists.txt

```diff
@@ -291,11 +291,36 @@ endif ()
 
 # pybinds
 if (ENABLE_PYBINDS)
+    if(DEFINED PYTHON_EXECUTABLE)
+        message(STATUS "Using specified Python executable: ${PYTHON_EXECUTABLE}")
+        set(Python3_EXECUTABLE ${PYTHON_EXECUTABLE})
+    elseif(DEFINED Python_EXECUTABLE)
+        message(STATUS "Using specified Python executable: ${Python_EXECUTABLE}")
+        set(Python3_EXECUTABLE ${Python_EXECUTABLE})
+    elseif(DEFINED Python3_EXECUTABLE)
+        message(STATUS "Using Python3_EXECUTABLE: ${Python3_EXECUTABLE}")
+    else()
+        message(WARNING "No Python executable specified, using system default")
```

Comment:
Q: what is the difference between `PYTHON_EXECUTABLE`, `Python_EXECUTABLE`, and `Python3_EXECUTABLE`?